zgw/object-management/src/main/kotlin/com/ritense/objectmanagement/service/ObjectManagementService.kt

```diff
@@ -98,16 +104,174 @@ class ObjectManagementService(
+        return objectsList.results.map {objects ->
+            val listRowDto = listColumns?.map {
+                listColumn ->
+                ObjectsListRowDto.ObjectsListItemDto(
+                    listColumn.key,
+                    objects.record.data?.at(listColumn.path)
+                )
+            }
+            ObjectsListRowDto(objects.uuid.toString(), listRowDto!!)
+        }
+    }
+
+    private fun mapToObjectSearchParameter(
+        key: String,
+        fieldType: FieldType,
+        value: Any,
+        dataType: DataType
+    ): List<ObjectSearchParameter> {
+        return when (fieldType) {
+            FieldType.TEXT_CONTAINS -> listOf(
+                ObjectSearchParameter(
+                    key,
+                    Comparator.STRING_CONTAINS,
+                    castValueToDataType(dataType, value)
+                )
+            )
+
+            FieldType.RANGE -> {
+                (value as List<Any?>).mapIndexed { index, rangeValue ->
+                    if (rangeValue != null) {
+                        ObjectSearchParameter(
+                            key,
+                            if(index == 0) Comparator.GREATER_THAN_OR_EQUAL_TO else Comparator.LOWER_THAN_OR_EQUAL_TO,
+                            castValueToDataType(dataType, rangeValue)
+                        )
+                    } else {
+                        null
+                    }
+                }.filterNotNull()
+            }
+
+            FieldType.MULTI_SELECT_DROPDOWN -> {
+                value as List<String>
+                return value.map {
+                    ObjectSearchParameter(key, Comparator.EQUAL_TO, it)
+                }
+            }
+
+            FieldType.SINGLE_SELECT_DROPDOWN -> listOf(
+                ObjectSearchParameter(key, Comparator.EQUAL_TO, castValueToDataType(dataType, value))
+            )
+
+            FieldType.SINGLE -> listOf(
+                ObjectSearchParameter(key, Comparator.STRING_CONTAINS, castValueToDataType(dataType, value))
+            )
+        }
+    }
+
+    fun castValueToDataType(dataType: DataType, value: Any): String {
+        return when (dataType) {
+            DataType.TEXT -> value as String
+            DataType.NUMBER -> {
+                value as Number
+                return value.toString()
+            }
+
+            DataType.BOOLEAN -> {
+                value as Boolean
+                return value.toString()
+            }
+
+            DataType.DATE -> {
+                value as Date
+                return value.toString()
+            }
+
+            else -> {
+                value as Time
+                return value.toString()
+            }
+        }
+
+    }
+
+    private fun concatenateObjectSearchParameter(searchParameterList: List<ObjectSearchParameter>): String {
+        return searchParameterList.joinToString(separator = ",") { it.toQueryParameter() }
```

Comment:
This same operation is also defined in `ObjectSearchParameter.toQueryParameter(list)`